zap/src/main/java/org/zaproxy/zap/extension/anticsrf/ExtensionAntiCSRF.java

```diff
@@ -492,7 +492,26 @@ public String generateForm(int hrefId) throws Exception {
      * @since 2.7.0
      */
     public String generateForm(HttpMessage msg) throws UnsupportedEncodingException {
+        return generateForm(msg, new HashMap<String, String>());
+    }
+
+    /**
+     * Generates a HTML form from the given message.
+     *
+     * @param msg the message used to generate the HTML form, must not be {@code null}.
+     * @param optionalParams optional parameters for form
```

Comment:
Why pass a hashMap when all that's necessary is a string?

Reply:
Just thinking if there will be more parameters, we can just add them to the map.

Reply:
I'll let @psiinon, @thc202, or @ricekot chime in but I think it's best to keep it to what's needed.

Reply:
We shouldn't use a map for that purpose though, it's fragile. Just a string is fine, the method does not need to be public.

Reply:
fixed